server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -321,7 +321,7 @@ public final BytesReference requiredContent() {
         if (hasContent() == false) {
             throw new ElasticsearchParseException("request body is required");
         } else if (xContentType.get() == null) {
-            throw new IllegalStateException("unknown content type");
+            throw new IllegalArgumentException("unknown content type");
```

Comment:
I see why you are making the change, but technically the IllegalStateException here was correct (since xContent is a field, not an argument).
Do you think this check could be moved at the time xContent is set (the ctor I think?).
That would make the IllegalArgumentException "more legit".
Alternatively, what about using one of our own exception types (ElasticsearchParseException, or a new one like XContentException)?

Reply:
> Moving to the ctor is not straight foward either because this is a static utility method that is used elsewhere.

That was my fear too. It would have been more fitting to throw a IAE where the argument is actually processed, but if it's not trivial, better not do it.

>  IllegalStateException also doesn't sound particularly fitting for this case

Yeah, reading the docs it does not seem like it's fitting - I thought more of an exception like "your class invariants are broken" or "the various fields in your class do not play well in this case".

TBH `ValidationException` seems better to me, but I'm fine either way.

Reply:
👍 I changed to ValidationException, hope I don't need to adapt more tests to get through CI